.github/workflows/tests-nightly.yml

```diff
@@ -0,0 +1,67 @@
+name: Run tests nightly
+on:
+  workflow_dispatch: {}
+  schedule:
+    - cron: '0 5 * * *' # 5AM UTC = 12AM EST
+   # TODO: remove this pull_request section before merging. it's just for testing this GH action pre-merge
+  pull_request:
+    branches:
+      - '**'
+
+jobs:
+  tests-against-source-code-and-latest-install:
+    strategy:
+      matrix:
+        testOptions: [ "-PtestTag=unit", "-PtestTag=integration", "-PtestTag=integration -PtestInstallFromGitHub" ]
+      fail-fast: false
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout current code
+        id: checkout_code
+        uses: actions/checkout@v2
+        with:
+          token: ${{ secrets.BROADBOT_GITHUB_TOKEN }}
+      - name: Set up AdoptOpenJDK 11
+        id: setup_jdk
+        uses: joschi/setup-jdk@v2
+        with:
+          java-version: 11
+      - name: Cache Gradle packages
+        id: cache_gradle
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: v1-${{ runner.os }}-gradle-${{ hashfiles('**/gradle-wrapper.properties') }}-${{ hashFiles('**/*.gradle') }}
+          restore-keys: v1-${{ runner.os }}-gradle-${{ hashfiles('**/gradle-wrapper.properties') }}
+      - name: Grant execute permission for gradlew
+        id: allow_execute_gradle
+        run: chmod +x gradlew
+      - name: Render config
+        id: render_config
+        run: |
+          # this step does the equivalent of the tools/render-config.sh script.
+          # on local machines, the script fetches a SA from Vault.
+          # in GH actions, the SA key is stored in a GH repo secret.
+          # regardless of how it was fetched, tests expect this
+          # key to be stored in rendered/test-user-account.json
+          mkdir -p rendered
+          echo "$TEST_USER_SA_KEY" > rendered/test-user-account.json
+        env:
+          TEST_USER_SA_KEY: ${{ secrets.TEST_USER_SA_KEY }}
```

Comment:
not this PR, but in the past dev ops has asked us to keep secrets like test accounts in Vault and only store Vault tokens as github secrets. Then, the GHA uses the vault token to get the service account secret from vault. This helps when rotating credentials. It also makes it easy to access more secrets as needed.
We also get the advantage of being able to run `tools/render-config.sh` script directly. e.g. buffer: https://github.com/DataBiosphere/terra-resource-buffer/blob/df45bd5c4ab163e7651d930529092cfaf07281fc/.github/workflows/test.yml#L35-L49
I found getting a pattern to be a little bit of "which DevOps engineer happened to talk to you" but consider filing a ticket to follow up with the Vault pattern for the CLI.

Reply:
I specifically asked DevOps about this and they told me to put the secret as a GH secret instead of accessing Vault directly ([slack conversation](https://broadinstitute.slack.com/archives/CADM7MZ35/p1615817519243700)). So definitely seems to be dependent on who you talk to.

I'd prefer to have an agreed upon central approach before making any changes here. Being able to use the same `render-config` script would be good. Also, Dan told me that they recently changed the Artifactory password in Vault, which broke lots of GH actions that had copy/pasted the value into a secret. That's another possible reason to move away from this model, or have a list somewhere of which GH secrets need to be updated, etc.